lib/database/model/event_model.dart

```diff
@@ -19,57 +21,135 @@ class EventModel {
   late List<String>? attachments;
   late String? imageUrl;
   late String? organizer;
+  late String? type;
   late bool registered;
 
-  EventModel(
-      this.id,
-      this.eventID,
-      this.title,
-      this.startDateAndTime,
-      this.endTime,
-      this.location,
-      this.description,
-      this.contactName,
-      this.contactEmail,
-      this.eventUrl,
-      this.durationDays,
-      this.attachments,
-      this.imageUrl,
-      this.organizer,
-      this.registered);
-
-  factory EventModel.fromEvent(Event event, bool registered) {
-    return EventModel(
-        int.parse(event.eventID),
-        event.eventID,
-        event.title,
-        event.startDateAndTime,
-        event.endTime,
-        event.location,
-        event.description,
-        event.contactName,
-        event.contactEmail,
-        event.eventUrl,
-        event.durationDays,
-        event.attachments,
-        event.imageUrl,
-        event.organizer,
-        registered);
+  DateTime? get endDate =>
+      startDateAndTime?.add(Duration(days: durationDays ?? 1 - 1));
+
+  String? get locationName {
+    if (location.isNotNullAndNotEmpty) {
+      final parts = location!.split(":");
+      return parts[0];
+    }
+    return null;
+  }
+
+  String? get locationForMap {
+    if (location.isNotNullAndNotEmpty) {
+      final parts = location!.split(":");
+      if (parts.length > 1) {
+        return parts[1].trim();
+      }
+    }
+    return null;
   }
 
-  Event toEvent() {
-    return Event(
+  static RegExp regexStartTime = RegExp(r'^(?:[01]\d|2[0-3]):[0-5]\d$');
+  static RegExp regexStartAndEndTime = RegExp(r'^\d{2}:\d{2}-\d{2}:\d{2}$');
+
+  EventModel({
+    required this.eventID,
+    required this.title,
+    this.startDateAndTime,
+    this.endTime,
+    this.location,
+    this.description,
+    this.contactName,
+    this.contactEmail,
+    this.eventUrl,
+    this.durationDays,
+    this.attachments,
+    this.imageUrl,
+    this.organizer,
+    this.type,
+    this.registered = false,
+  }) : id = int.parse(eventID);
+
+  static EventModel? fromJson(Map<String, dynamic> json) {
+    // match necessary json values
+    if (json
+        case {
+          'id': String eventID,
+          'zeit': String timeInput,
+          'titel': String titel,
+          'datum': String datum,
+        }) {
```

Comment:
Dart 3: pattern matching

Nur wenn die 4 keys existieren und Strings sind kommt man in das if. Gleichzeitig werden die Werte auch in den Variablen gespeichert.

Reply:
Lit 🔥